### PR TITLE
feat(pack): Add ruby language pack

### PR DESCRIPTION
**Pull request by dsisnero**

### CODE REVIEW

**Comment by Uzaaft** [comment]
@dsisnero Could you revert [4d95c29](https://github.com/AstroNvim/astrocommunity/pull/323/commits/4d95c299b83a64578cac87e08029a37bc691423f), and instead open another PR for that?

**Comment by dsisnero** [comment]
done

File: lua/astrocommunity/pack/ruby/init.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    opts = function(_, opts)
+      if opts.ensure_installed ~= "all" then
+        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "ruby", "html" })
+      end
+    end,
+  },
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts)
+      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "solargraph", "ruby_ls" })
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/ruby/init.lua:15]: Wouldn't the two LSP's cause conflicts with each other?

**Comment by dsisnero** [comment]
removed html and just installed solargraph lsp - will add another option to use ruby_ls later

File: lua/astrocommunity/pack/ruby/init.lua [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  {
+    "jay-babu/mason-null-ls.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "standardrb") end,
+  },
+  {
+    "mfussenegger/nvim-dap",
+    dependencies = {
+      -- "suketa/nvim-dap-ruby",
+      "takkii/neoruby-debugger",
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/ruby/init.lua:24]: neoruby-debugger isn't a github repo

**Comment by Uzaaft** [comment]
@dsisnero  I couldn't find the neoruby package you're using

**Comment by Uzaaft** [comment]
> @dsisnero I couldn't find the neoruby package you're using

My bad it just didn't appear in google. Had to dig for it.

File: lua/astrocommunity/pack/ruby/init.lua [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      -- config = function() require("dap-ruby").setup() end,
+      config = function() require("neoruby-debugger").setup() end,
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/ruby/init.lua:26]:
```suggestion
      "suketa/nvim-dap-ruby",
      config = function() require("dap-ruby").setup() end,
```
Not sure if we should have neoruby in the pack when most of our users wouldn't be able to use their docs.

Comment (dsisnero) [reply lua/astrocommunity/pack/ruby/init.lua:26]: It doesn't really need docs.  Makes nvim-dap just work for ruby.

Comment (Uzaaft) [reply lua/astrocommunity/pack/ruby/init.lua:26]: It does. For example the docs mention that Ruby 3.2.x apparently doesn't build. I'd vote against it.

Comment (Uzaaft) [reply lua/astrocommunity/pack/ruby/init.lua:26]: It'll be too tricky to maintain a pack which has non-english docs, if we were to get questions about it, just like we do with other plugins. And the plugin itself doesn't seem to be that used(low amount of stars, and few people watching it.

Comment (Uzaaft) [reply lua/astrocommunity/pack/ruby/init.lua:26]: @dsisnero  friendly reminder

**Comment by Uzaaft** [comment]
I went ahead and commiten the non-neoruby dap.

**Review by luxus — APPROVED** [review]

File: lua/astrocommunity/pack/ruby/init.lua [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  {
+    "jay-babu/mason-null-ls.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "standardrb") end,
+  },
+  {
+    "mfussenegger/nvim-dap",
+    dependencies = {
+      "suketa/nvim-dap-ruby",
+      config = function() require("dap-ruby").setup() end,
```
Comment (mehalter) [inline lua/astrocommunity/pack/ruby/init.lua:24]:
```suggestion
      "suketa/nvim-dap-ruby",
      name = "dap-ruby",
      opts = {},
```

File: lua/astrocommunity/pack/ruby/init.lua [hunk 5]
```diff
@@ -0,0 +1,27 @@
+local utils = require "astronvim.utils"
+
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if opts.ensure_installed ~= "all" then
+        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "ruby" })
```
Comment (mehalter) [inline lua/astrocommunity/pack/ruby/init.lua:8]:
```suggestion
        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "ruby")
```

**Review by mehalter — CHANGES_REQUESTED** [review]

File: lua/astrocommunity/pack/ruby/init.lua [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if opts.ensure_installed ~= "all" then
+        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "ruby")
+      end
+    end,
+  },
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "solargraph" }) end,
```
Comment (mehalter) [inline lua/astrocommunity/pack/ruby/init.lua:14]:
```suggestion
    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "solargraph") end,
```

**Review by mehalter — CHANGES_REQUESTED** [review]
Oh I totally missed one more copy edit. Sorry! But this looked great :D

**Review by luxus — APPROVED** [review]

**Pull request merged**